app/models/facility.rb

```diff
@@ -239,6 +240,18 @@ def facility_is_unique
     errors.add(:facility, "already exists") if organization.present? && facility_group.present? && facility.present?
   end
 
+  def block_allowed
```

Comment:
Replace this code snippet with:

```ruby
  def block_allowed
    unless facility_group.region.blocks.pluck(:name).include?(block)
      errors.add(:zone, "not present in the facility group")
    end
```

...once https://github.com/simpledotorg/simple-server/pull/1554 is merged in.